PencilCase/PencilCase/scripts/Pages/ImportProduct.ts

```diff
@@ -0,0 +1,173 @@
+﻿import { PageBase } from './PageBase';
+import { Navigator } from '../Navigator';
+import * as Consts from './Consts';
+import { Product } from '../Models/Product';
+import { ProductRepository } from '../Repositories/ProductRepository';
+import { OrderRepository } from '../Repositories/OrderRepository';
+import { Order } from '../Models/Order';
+import * as Utils from '../Utils';
+import { OrderTypes } from '../Models/Order';
+
+export class ImportProduct extends PageBase {
+    private navigator: Navigator = Navigator.instance;
+    private products: KnockoutObservableArray<Product> = ko.observableArray([]);
+    private dict = {};
+    private selectOptions = { Id: "placeholder", Name: "选择产品……", WholesalePrice: 0, WholesaleUnitName: '' };
+    private selectedProductId: KnockoutObservable<string> = ko.observable(this.selectOptions.Id);
+    private selectedProduct: KnockoutObservable<any> = ko.observable(this.selectOptions);
+    private productSelected = ko.computed(() => {
+        if (this.selectedProduct() && this.selectedProduct().Id !== this.selectOptions.Id)
+            return true;
+        return false;
+    });
+
+    private selectedProductQuantity: KnockoutObservable<number> = ko.observable(1);
+    private selectedProductImportPrice: KnockoutObservable<number> = ko.observable(0);
+    private clearSelection = ko.observable(false);
+
+    private importOrders = ko.observableArray<Order>([]);
+    private batchId: string = null;
+    private totalNumber: KnockoutObservable<number> = ko.observable(0);
+    private totalPrice: KnockoutObservable<number> = ko.observable(0);
+
+    private productRepository = new ProductRepository();
+    private orderRepository = new OrderRepository();
+
+    constructor() {
+        super();
+        this.title = ko.observable("进货");
+        this.pageId = Consts.Pages.ImportProduct.Id;
+        this.back = Navigator.instance.goHome;
+        this.selectedProductId.subscribe((newValue: string) => {
+            this.selectedProduct(this.dict[newValue]);
+            if (this.selectedProduct() && this.selectedProduct().Id !== this.selectOptions.Id) {
+                this.selectedProductImportPrice(this.selectedProduct().ImportWholesalePrice ? this.selectedProduct().ImportWholesalePrice : 0);
+            }
+        });
+    }
+
+    public initialize() {
+        let productRepository: ProductRepository = new ProductRepository();
+        productRepository.getAll((transaction: SqlTransaction, resultSet: SqlResultSet) => {
+            this.products([]); // First, clear products collection
+            let rows = resultSet.rows;
+            let array = [];
+            this.dict = {};
+            for (let i = 0; i < rows.length; i++) {
+                //this.products.push(new Product(rows.item(i)));
+                array.push(new Product(rows.item(i)));
+                this.dict[(<any>(rows.item(i))).Id]= rows.item(i);
```

Comment:
Create this dictionary, so it's will easy to find the corresponding product once selected product id changed. Because the selected value binding can only be one field of the product but not the entire product instance.